scripts/templates/index.d.ts.tpl

```diff
@@ -32,11 +32,8 @@ declare namespace Github {
   }
 
   export interface Options {
+    baseUrl?: string;
     timeout?: number;
-    host?: string;
-    pathPrefix?: string;
-    protocol?: string;
-    port?: number;
```

Comment:
I think I would leave these, they are deprecated but still supported. That way we don’t break existing code.

Is there a way to mark something as "deprecated" in Typescript?

Reply:
I think one way is to [decorate](http://usejsdoc.org/tags-deprecated.html) it with `@deprecated` so that you get the message from the language server. Not sure if this causes a warning in the build.

Reply:
Mind if I update your PR? I’ve extended the typescript test to make it less likely that I break it again :(

Reply:
Be my guest!

Reply:
JSDoc doesn't cause a build failure, but it does do this:

<img width="974" alt="image" src="https://user-images.githubusercontent.com/13207348/37497192-1f97942e-288d-11e8-9f7e-e87c5f14c7be.png">

Reply:
I don’t use JS doc in this repo, is this an editor plugin?

Reply:
>  is this an editor plugin?

It's part of the VS Code / Atom language server for TypeScript

Reply:
okay thanks, I added them to each deprecated option